src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/Handlers/AbstractLanguageServiceRuleHandler.cs

```diff
@@ -1,25 +1,23 @@
 ﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
-using System.Threading.Tasks;
 using Microsoft.VisualStudio.LanguageServices.ProjectSystem;
 
 namespace Microsoft.VisualStudio.ProjectSystem.LanguageServices.Handlers
 {
-    internal abstract class AbstractLanguageServiceRuleHandler : ILanguageServiceRuleHandler
+    internal abstract class AbstractLanguageServiceRuleHandler : IEvaluationHandler
     {
         public abstract RuleHandlerType HandlerType { get; }
 
-        public abstract string RuleName { get; }
+        public abstract string EvaluationRuleName { get; }
 
         public virtual bool ReceiveUpdatesWithEmptyProjectChange => false;
 
         public virtual void Handle(IProjectChangeDescription projectChange, IWorkspaceProjectContext context, bool isActiveContext)
         {
         }
 
-        public virtual Task OnContextReleasedAsync(IWorkspaceProjectContext context)
+        public virtual void OnContextReleasedAsync(IWorkspaceProjectContext context)
```

Comment:
Are these still supposed to be Async? If so, why async void? If not, remove Async suffix.